### PR TITLE
fix: make credit actions disable on masquerade

### PR DESCRIPTION
**Pull request by leangseu-edx**
disable credit action while masquerading.

https://2u-internal.atlassian.net/browse/AU-1091

Screenshot 2023-04-17 at 12 44 02 PM
Screenshot 2023-04-17 at 12 44 28 PM
Screenshot 2023-04-17 at 12 45 08 PM

### CODE REVIEW

**Review by nsprenkle — APPROVED** [review]
This looks good, but we should also make it clear to support that these are disabled *intentionally*. Either by:
1. Signifying in the docs that these actions are intentionally disabled OR
2. (My preference, for later) show these as they would appear to a user but disallow clicking by masqueraded staff (e.g. show the `not-allowed` cursor on hover). This gets us closer to a "masquerade" experience for troubleshooting but disallows dangerous actions.

File: src/containers/CourseCard/components/CourseCardBanners/CreditBanner/views/ApprovedContent.jsx [hunk 1]
```diff
@@ -11,10 +11,11 @@ import messages from './messages';
 
 export const ApprovedContent = ({ cardId }) => {
   const { providerStatusUrl: href, providerName } = reduxHooks.useCardCreditData(cardId);
+  const { isMasquerading } = reduxHooks.useMasqueradeData();
   const { formatMessage } = useIntl();
   return (
     <CreditContent
-      action={{ href, message: formatMessage(messages.viewCredit) }}
```
Comment (jansenk) [inline src/containers/CourseCard/components/CourseCardBanners/CreditBanner/views/ApprovedContent.jsx:17]: any reason this is here and not as its own prop?

File: src/containers/CourseCard/components/CourseCardBanners/CreditBanner/views/ApprovedContent.jsx [hunk 2]
```diff
@@ -11,10 +11,11 @@ import messages from './messages';
...
+      action={{ href, message: formatMessage(messages.viewCredit), disabled: isMasquerading }}
```
Comment (jansenk) [inline src/containers/CourseCard/components/CourseCardBanners/CreditBanner/views/ApprovedContent.jsx:18]: Any reason this is here and not its own prop?

**Review by jansenk — APPROVED** [review]

File: src/containers/CourseCard/components/CourseCardBanners/CreditBanner/views/ApprovedContent.jsx (hunk 2 above)
Comment (leangseu-edx) [reply src/containers/CourseCard/components/CourseCardBanners/CreditBanner/views/ApprovedContent.jsx:18]: my thought process was that disable is part of actions. If action is undefined, disable doesn't mean anything anyway.

**Pull request merged**